addons/website/models/website.py

```diff
@@ -312,6 +312,15 @@ def create_and_redirect_configurator(self):
         configurator_action_todo = self.env.ref('website.website_configurator_todo')
         return configurator_action_todo.action_launch()
 
+    def _is_indexable_url(self, url):
+        """
+        Returns True if the given url has to be indexed by search engines.
```

Comment:
Maybe it's worth something more than this (but a bit less detailed than commit msg) explaining that the website domain and the url "domain" should match to be considered indexable.
And that matching means being equal once scheme and potential `www` has been removed.
That this is to be able to still index badly configured domain (when the domain is not the real canonical url due to www/naked or http/https redirect).

Probably not worth mentioning the tradeoff: if a website is reachable both in www+naked or http+https, it might be indexed twice but it's still better than preventing completely the indexation of a website if the domain is misconfigured (not the real endpoint).

Because I am sure no one will understand this by reading the code.